libcxx/test/support/test_macros.h

```diff
@@ -262,6 +262,10 @@
 
 #define TEST_IGNORE_NODISCARD (void)
 
+#if !(defined(_LIBCPP_VERSION) && !_LIBCPP_AVAILABILITY_HAS_FROM_CHARS_FLOATING_POINT)
```

Comment:
```suggestion
#if !defined(_LIBCPP_VERSION) || _LIBCPP_AVAILABILITY_HAS_FROM_CHARS_FLOATING_POINT)
```
IMO this is significantly easier to read.